### PR TITLE
Fix bug where vscode extensions are not installing

### LINKED ISSUE #1350: [BUG] VS Code extensions are not installing
### OS system and architecture in which you are running QHub

GCP

### Expected behavior

* VS Code extensions should be installable by users
* python extension should be installed by default

### Actual behavior

* Trying to install an extension fails

### How to Reproduce the problem?

1. Attempt to install any extension from VS Code Extensions

### Command output

```bash session
[2022-06-28 10:07:41.756] [window] [info] [Service Worker] registered
[2022-06-28 10:07:41.892] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)"]
[2022-06-28 10:07:41.982] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)"]
[2022-06-28 10:07:42.257] [window] [error] Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions': Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'
    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919
    at async Promise.all (index 1)
    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)
[2022-06-28 10:07:45.763] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)"]
[2022-06-28 10:07:45.945] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)"]
[2022-06-28 10:07:46.001] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)"]
[2022-06-28 10:07:46.030] [window] [error] Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions': Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'
    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919
    at async Promise.all (index 0)
    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)
[2022-06-28 10:08:07.310] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)"]
[2022-06-28 10:08:09.031] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)"]
[2022-06-28 10:08:16.412] [window] [error] ["Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'","    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919","    at async Promise.all (index 1)","    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)","    at async U.getAllDepsAndPackExtensionsToInstall (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:108:9171)","    at async U.installExtension (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:108:5108)","    at async U.installFromGallery (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:108:3035)"]
[2022-06-28 10:08:16.439] [window] [error] Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions': Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'
    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919
    at async Promise.all (index 1)
    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)
    at async U.getAllDepsAndPackExtensionsToInstall (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:108:9171)
    at async U.installExtension (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:108:5108)
    at async U.installFromGallery (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:108:3035)
[2022-06-28 10:08:16.526] [window] [error] Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions': Internal: Unable to resolve nonexistent file '/opt/conda/envs/default/share/code-server/extensions'
    at /opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:78919
    at async Promise.all (index 1)
    at async i.scanExtensions (/opt/conda/envs/default/share/code-server/lib/vscode/out/vs/server/node/server.main.js:146:79033)
```

* `/opt/conda/envs/default/share/code-server/extensions` doesn't exist. 
* `/opt/conda/envs/default/share/code-server/` exists, but my user doesn't have permission to create a new folder herer
```


### Versions and dependencies used.

_No response_

### Compute environment

GCP

### Integrations

_No response_

### Anything else?

Something may have changed recently because I think extensions used to be stored in `~/.local/share/code-server/extensions`

### PR DESCRIPTION
**Pull request by viniciusdc**
Fixes | Closes | Resolves #1350 

## Changes introduced in this PR:

- This PR switched back the installation for the code-server backend to how it was previously provided before we moved to the conda-forge `code-server` feedstock.
- Due to an issue with the docker build for the new image, which seems to be node related, I needed to bump the installation for the dask-labextension to the latest version as well 
  - This includes the fixes proposed for #1327 and adds a new re-branding for dask 
  - The extensions for VScode as well its main config are now presented in `/home/<user>/.local/shared/code-server` dir

## Types of changes

What types of changes does your PR introduce?

_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

### Requires testing

- [x] Yes
- [ ] No

### In case you checked yes, did you write tests?

- [ ] Yes
- [x] No
- [ ] 
## Further comments (optional)

This should count as a quick fix for the current extension installation issue of VScode, but we need to revisit this after the current sprint.

### CODE REVIEW

**Comment by viniciusdc** [comment]
The previous extension mechanism is preserved 
image

Also, the extension addition mechanism is now working:
image

from the Terminal:
```bash
test-user@jupyter-test-2duser:~/.local/share/code-server$ ls
CachedExtensionVSIXs  CachedExtensions  Machine  User  coder-logs  coder.json  heartbeat  logs  machineid
test-user@jupyter-test-2duser:~/.local/share/code-server$ 
```

**Comment by viniciusdc** [comment]
I just need to test the `dask-labextension` to make sure it dint break anything

**Comment by viniciusdc** [comment]
Dask-labextension is now fixed:
image

**Comment by viniciusdc** [comment]
It seems that fixing the extensions broke the terminal sync with the Jupyterlab terminal (we now get a blank terminal on vscode), trying to fix that.

File: qhub/template/image/jupyterlab/environment.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ dependencies:
  - retrolab
 
  # jupyterlab extensions
-#  - dask_labextension >=5.0.0
+ - dask_labextension == 5.3.0
```
Comment (costrouc) [inline qhub/template/image/jupyterlab/environment.yaml:22]: Can we make this a forward pin to `>=5.3.0`?

**Review by costrouc — CHANGES_REQUESTED** [review]

**Comment by costrouc** [comment]
> It seems that fixing the extensions broke the terminal sync with the Jupyterlab terminal (we now get a blank terminal on vscode), trying to fix that.

These components don't seem related not sure why this would be happening.

File: qhub/template/image/jupyterlab/environment.yaml (hunk 1 above)
Comment (viniciusdc) [reply qhub/template/image/jupyterlab/environment.yaml:22]: no problem, I just updated because the yarn version in there was breaking the ci

**Comment by viniciusdc** [comment]
> > It seems that fixing the extensions broke the terminal sync with the Jupyterlab terminal (we now get a blank terminal on vscode), trying to fix that.
> 
> These components don't seem related not sure why this would be happening.

seems to be a node thing, the terminal is in there but it's not rendered correctly -- I am trying to change the /bin path to point directly to the `code-server + node` installation (as we have on quansight-beta), I also update the order of calling the code-server prosy to use the conda env (I think it was being installed in the pip env of the docker image and not the default)

this is the error, for ref.:
```
[2022-07-06 12:40:58.929] [window] [error] [object Event]: Error: [object Event]
    at Object.E [as ensureError] (https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:3:4182)
    at i._createLoadError (https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:5:8131)
    at i._onLoadError (https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:5:8250)
    at f (https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:5:10089)
    at Object.errorback (https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:5:10445)
    at e.triggerErrorback (https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:3:11144)
    at https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:3:10850
    at HTMLScriptElement.f (https://gcpqhubstages.qhub.dev/user/test-user/vscode/stable-30d9c6cd9483b2cc586687151bcbcd635f373630/static/out/vs/loader.js:3:11365)
```

**Review by iameskild — APPROVED** [review]
Great work @viniciusdc :)

**Pull request merged**